docs/source/architecture/journal.rst

```diff
@@ -0,0 +1,379 @@
+
+In the spirit of configurability, the Journal supports pluggable consensus
+algorithms that may be changed via the Configuration transaction family.  The
+selected Consensus algorithm is set for the chain in the Genesis Block during
+Genesis (described below). The Journal and its Consensus Interface support
+pluggable consensus for probabilistic finality algorithms like Proof of Work,
+as well as algorithms with absolute finality like PBFT.
+
+The Consensus algorithm services to the journal are divided into three distinct
+interfaces that have specific lifetimes and access to information.
+
+1. Consensus.BlockPublisher
+2. Consensus.BlockVerifier
+3. Consensus.ForkResolver
+
+Consensus algorithm implementations in Sawtooth Lake must implement all of the
+consensus interfaces. Each of these objects are provided read-only access to
+the BlockCache and GlobalState.
+
+Consensus.BlockPublisher
+------------------------
+
+The Consensus.BlockPublisher implementation of this interface is used by the
+BlockPublisher to create new candidate blocks to extend the chain.  The block
+publisher is provided access to the a read-only view of global state, a
+read-only view of the BlockStore, and an interface to publish batches.
+
+Three events are called on the Consensus.BlockPublisher, 
+
+1. initialize_block - The BlockHeader is provided for the candidate block. This
+   is called immediately after the block_header is initialized and allows for
+   validation of the consensus algorithm's internal state, checks if the header
+   is correct, checks if according to the consensus rules a block could be
+   published, and checks if any initialization of the block_header is required.
+   If this function fails no candidate block is created and the BlockPublisher
+   will periodically attempt to create new blocks.
+2. check_publish_block - Periodically, polling is done to check if the block can
+   be published. In the case of PoET, this is a check to see if the wait time
+   has expired, but could be on any other criteria the consensus algorithm has
+   for determining if it is time to publish a block. When this returns true the
+   BlockPublisher will proceed in creating the block.
+3. finalize_block - Once check_publish_block has confirmed it is time to
+   publish a block, the block header is considered complete, except for the
+   consensus information. The BlockPublisher calls finalize_block with the
+   completed block_header allowing the consensus field to be filled out.
+   Afterwards, the BlockPublisher signs the block and broadcasts it to the
+   network.
+
+This implementation needs to take special care to handle the genesis block
+correctly. During genesis operation, the Consensus.BlockPublisher will be called
+to initialize and finalize a block so that it can be published on the chain
+(see below).
+
+Consensus.BlockVerifier
+-----------------------
+
+The Consenus.BlockVerifier implementation Provides Block verification services
```

Comment:
Spurious capitalization of "Provides"? "The Consenus.BlockVerifier implementation **Provides** Block verification services"